### ISSUE TITLE
Keychain dialogue now shows on app launch?

### ISSUE THREAD

**Issue by jessegrosjean** [PaddleHQ/Mac-Framework-V4#44]
I don't remember this happening before, but now when I launch the paddle version of my app I see a keychain access dialogue asking me to enter my "login" keychain password:

<img width="467" alt="screen shot 2019-01-09 at 10 13 14 am" src="https://user-images.githubusercontent.com/21868/50908187-32b3ba00-13f7-11e9-9aa3-22b14e574023.png">

Is this expected? I really don't think it's good user experience. In fact it shows twice on each launch.

**Comment by jessegrosjean** [comment]
Thanks for your quick response.

I have for a long time signed my outside of App Store version of TaskPaper with Developer ID. When I did the build this time Xcode did download a new version of my certificates for me (I have no idea on details, I just have it set to manage keys/certificates automatically).

My signing info reported by RB App Checker looks like this:

```
Evaluating the application “TaskPaper Direct”.

The application was signed by “Apple Root CA”, “Developer ID Application: Jesse Grosjean (64A5CLJP5W)”.
	Both the verified timestamp and the signing-time are: Jan 7, 2019 at 11:37:36 AM.
	The object code format is “app bundle with Mach-O thin (x86_64)”.
	The signature contains the Team ID “64A5CLJP5W”.
	Both bundle and signing identifiers are “com.hogbaysoftware.TaskPaper3.direct”.
	The signature specifies implicit requirements. ￼
		The requirements specify the Team ID “64A5CLJP5W”.
			This matches the Team ID contained in the signature.
	The signature specifies resource rules (v1). ￼
	The signature specifies resource rules (v2). ￼
	Requirements and resources validate correctly.

The code signature has the UUID “96BBE396-FEB6-DB7B-0B62-0506DD17B7D5”.
	Executable code for x86_64 has the UUID “2A381DF9-0FE5-3FC1-9C94-F069023F98B8”.

A signing-time snapshot of the application’s Info.plist was found. ￼
	Version 3.8.3 (338) Copyright © 2005–2018 Jesse Grosjean. All rights reserved.

The signature contains 3 certificates. ￼
	Certificate “Apple Root CA”: ￼
		Your keychain contains this trusted root certificate.
		Will expire on Feb 9, 2035.
	Certificate “Developer ID Certification Authority”: ￼
		Will expire on Feb 1, 2027.
	Certificate “Developer ID Application: Jesse Grosjean (64A5CLJP5W)”: ￼
		Will expire on Feb 15, 2021.
		SHA1 fingerprint: “F8A1C0DD07F5B79F1732EDCA0CA07AFC5E980DFB”.
		Team ID or Organizational Unit: “64A5CLJP5W”.
			This matches the Team ID contained in the signature.

The application is probably from an authorized Apple Developer.

The code signature contains entitlements. ￼
		com.apple.security.get-task-allow: YES.
		com.apple.security.cs.allow-jit: YES.

The application is not sandboxed.

There are 4 embedded frameworks. ￼

22 auxiliary executables have been found. ￼
	22 executables are signed by  “Apple Root CA”, “Developer ID Application: Jesse Grosjean (64A5CLJP5W)”. ￼
```

**Comment by jessegrosjean** [comment]
> Use of the keychain should only be triggered when the app is signed with an Apple certificate, to avoid the dialogue box you are seeing.

So would you expect the dialogue to show for a developer ID signed app as shown (I think) above? I am under the strong impression that Apple wants all apps to be developer ID signed when distributed outside App Store.

**Comment by GregularExpressions** [comment]
@jessegrosjean just to confirm, you previously had an earlier version of TaskPaper installed using a different signing certificate?

The issue is that the keychain will consider the change in certificate another app, because we've stored some information at the app level, so it's prompting for access to read the entries locked to the previous app version and certificate.

Always Allow would cause this issue to go away (until the next certificate change) but each of your users in the same situation would be prompted.

We're not happy with that as a solution, we have some ideas we are bottoming out to avoid this completely. 

I should have another update today.

**Comment by jessegrosjean** [comment]
> @jessegrosjean just to confirm, you previously had an earlier version of TaskPaper installed using a different signing certificate?

Yes, I think so. I'm pretty sure what happened is the certificate that I used for TaskPaper 3.8.2 expired in late November. And so when I went to build TaskPaper 3.8.3 Xcode generated a new certificate and that's what TaskPaper 3.8.3 is signed with.

**Comment by GregularExpressions** [comment]
@jessegrosjean we have an approach that would prevent this in future, but it would require end users upgrading from SDK v4.0.6 or v4.0.7 to later SDK versions to see and accept the auth pop up that one time.

**Comment by mpaddle** [comment]
Hi @jessegrosjean,

We've prepared a beta that should address this issue. If you have some time, could you confirm that this new build fixes this issue?

[macOS-sdk-beta.zip](https://github.com/PaddleHQ/Mac-Framework-V4/files/2782816/macOS-sdk-beta.zip)

Kind regards,
Maarten

**Comment by GregularExpressions** [comment]
We believe this is fixed in [v4.0.8](https://github.com/PaddleHQ/Mac-Framework-V4/releases/tag/v4.0.8) 👍
